src/LinearMaps.jl

```diff
@@ -93,18 +94,39 @@ function LinearAlgebra.mul!(y::AbstractVector, A::LinearMap, x::AbstractVector,
         rmul!(y, β)
         return y
     else # α != 0, 1
-        iszero(β) && (A_mul_B!(y, A, x); rmul!(y, α); return y)
-        isone(β) && (y .+= rmul!(A * x, α); return y)
-        # β != 0, 1
-        rmul!(y, β)
-        y .+= rmul!(A * x, α)
+        iszero(β) && (mul!(y, A, x); rmul!(y, α); return y)
+        z = A * x
+        if isone(β)
+            y .+= z .* α
+        else
+            y .= y .* β .+ z .* α
+        end
         return y
     end
 end
+
 # the following is of interest in, e.g., subspace-iteration methods
-Base.@propagate_inbounds function LinearAlgebra.mul!(Y::AbstractMatrix, A::LinearMap, X::AbstractMatrix, α::Number=true, β::Number=false)
+@propagate_inbounds function mul!(Y::AbstractMatrix, A::LinearMap, X::AbstractMatrix)
     @boundscheck check_dim_mul(Y, A, X)
-    @inbounds @views for i = 1:size(X, 2)
+    return @inbounds _generic_mapmat_mul!(Y, A, X)
+end
+@propagate_inbounds function mul!(Y::AbstractMatrix, A::LinearMap, X::AbstractMatrix, α::Number, β::Number)
+    @boundscheck check_dim_mul(Y, A, X)
+    return @inbounds _generic_mapmat_mul!(Y, A, X, α, β)
+end
+
+@propagate_inbounds function _generic_mapmat_mul!(Y, A, X)
+    @views for i in 1:size(X, 2)
+        mul!(Y[:, i], A, X[:, i])
+    end
+    # starting from Julia v1.1, we could use the `eachcol` iterator
+    # for (Xi, Yi) in zip(eachcol(X), eachcol(Y))
+    #     mul!(Yi, A, Xi, α, β)
+    # end
+    return Y
+end
+@propagate_inbounds function _generic_mapmat_mul!(Y, A, X, α, β)
```

Comment:
How exactly? Having default values for alpha and beta? I think for some map types it's advantageous to keep 3-arg `mul!` calls as they are, say for `FunctionMap`s. I don't see how to do it exactly, but I'd happily accept a suggestion.

Reply:
I was thinking of default values, but it is definitely fine as is. However, even with default values, if `α = one(eltype(Y))` and `β = zero(eltype(Y))` (or just `true` and `false`), you would just end up calling 3-arg `mul!` of e.g. `FunctionMap` if the default values are used, no?

Reply:
True, it falls back to `_generic_mapvec_mul!`, which calls the 3-arg `mul!` if alpha is one and beta is zero. Agreed, let's merge them with default values. At some point I had lots of trouble with the default values, but the issue might have been something else. Now I'm very cautious with default values. It may well be that the conflicts were due to the fact that I had no "helper function", but only `mul!`s, and then I had infinite loops or ambiguities. As I read your first comment here, I guess you were spot-on.